From a5f1bc5f63ad17f434e2af8c95f3d0ff57b8b5c7 Mon Sep 17 00:00:00 2001
From: Bidipta Samadder <bidiptax.samadder@intel.com>
Date: Wed, 15 Jul 2015 18:20:24 +0530
Subject: [PATCH] [EM/BQ24292] Update VINDPM settings

VINDPM values need to be changed to
maintain sufficient gap between VINDPM threshold
and VBAT. This will avoid VBUS drop.
This change is tuned for both CHT-T4 and CHT-HR

Change-Id: I43e67443895ea2aaef3122f84329e6254c6a3ad1
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-21023
Signed-off-by: Bidipta Samadder <bidiptax.samadder@intel.com>
---
 drivers/power/bq24192_charger.c | 28 ++++++++++++++++++++--------
 1 file changed, 20 insertions(+), 8 deletions(-)

diff --git a/drivers/power/bq24192_charger.c b/drivers/power/bq24192_charger.c
index 02146dc0f5b2..cd521a31bccf 100644
--- a/drivers/power/bq24192_charger.c
+++ b/drivers/power/bq24192_charger.c
@@ -68,14 +68,17 @@
 #define INPUT_SRC_VOLT_LMT_404                 (1 << 4)
 #define INPUT_SRC_VOLT_LMT_412                 (3 << 3)
 #define INPUT_SRC_VOLT_LMT_444                 (7 << 3)
+#define INPUT_SRC_VOLT_LMT_452                 (1 << 6)
 #define INPUT_SRC_VOLT_LMT_460                 (9 << 3)
 #define INPUT_SRC_VOLT_LMT_468                 (5 << 4)
 #define INPUT_SRC_VOLT_LMT_476                 (0xB << 3)
 
 #define INPUT_SRC_VINDPM_MASK                  (0xF << 3)
-#define INPUT_SRC_LOW_VBAT_LIMIT               3600
-#define INPUT_SRC_MIDL_VBAT_LIMIT1             3800
-#define INPUT_SRC_MIDL_VBAT_LIMIT2             4100
+#define INPUT_SRC_LOW_VBAT_LIMIT               3400
+#define INPUT_SRC_MIDL_VBAT_LIMIT1             3600
+#define INPUT_SRC_MIDL_VBAT_LIMIT2             3800
+#define INPUT_SRC_MIDL_VBAT_LIMIT3             3900
+#define INPUT_SRC_MIDL_VBAT_LIMIT4             4100
 #define INPUT_SRC_HIGH_VBAT_LIMIT              4350
 
 /* D0, D1, D2 represent the input current limit */
@@ -1885,14 +1888,23 @@ static void bq24192_task_worker(struct work_struct *work)
 	 * Hence disabling dynamic vindpm update  for bq24296 chip.
 	*/
 	if (chip->chip_type != BQ24296) {
-		if (vbatt > INPUT_SRC_LOW_VBAT_LIMIT &&
-			vbatt <= INPUT_SRC_MIDL_VBAT_LIMIT1)
+		if (vbatt >= INPUT_SRC_LOW_VBAT_LIMIT &&
+			vbatt < INPUT_SRC_MIDL_VBAT_LIMIT1)
+			vindpm = INPUT_SRC_VOLT_LMT_404;
+		else if (vbatt >= INPUT_SRC_MIDL_VBAT_LIMIT1 &&
+			vbatt < INPUT_SRC_MIDL_VBAT_LIMIT2)
 			vindpm = INPUT_SRC_VOLT_LMT_412;
-		else if (vbatt > INPUT_SRC_MIDL_VBAT_LIMIT1 &&
-			vbatt <= INPUT_SRC_MIDL_VBAT_LIMIT2)
+		else if (vbatt >= INPUT_SRC_MIDL_VBAT_LIMIT2 &&
+			vbatt < INPUT_SRC_MIDL_VBAT_LIMIT3)
 			vindpm = INPUT_SRC_VOLT_LMT_444;
-		else if (vbatt > INPUT_SRC_MIDL_VBAT_LIMIT2)
+		else if (vbatt >= INPUT_SRC_MIDL_VBAT_LIMIT3 &&
+			vbatt < INPUT_SRC_MIDL_VBAT_LIMIT4)
+			vindpm = INPUT_SRC_VOLT_LMT_452;
+		else if (vbatt >= INPUT_SRC_MIDL_VBAT_LIMIT4 &&
+			vbatt < INPUT_SRC_HIGH_VBAT_LIMIT)
 			vindpm = INPUT_SRC_VOLT_LMT_460;
+		else
+			vindpm = INPUT_SRC_VOLT_LMT_468;
 
 		mutex_lock(&chip->event_lock);
 		ret = bq24192_modify_vindpm(vindpm);
-- 
1.9.1

